packages/trader/src/Modules/Trading/Components/Form/TradeParams/Multiplier/cancel-deal.jsx

```diff
@@ -18,6 +18,22 @@ const CancelDeal = ({
     toggleCancellationWarning,
 }) => {
     const should_show_popover = (has_take_profit || has_stop_loss) && should_show_cancellation_warning;
+    const [is_do_not_show_selected, setDoNotShowSelected] = React.useState(!should_show_cancellation_warning);
+    const is_do_not_show_ref = React.useRef(is_do_not_show_selected);
+
+    React.useEffect(() => {
+        is_do_not_show_ref.current = is_do_not_show_selected;
+    }, [is_do_not_show_selected]);
+
+    const onPopoverClose = React.useCallback(() => {
```

Comment:
I think if we have this func as normal callback we can get rid of `is_do_not_show_ref`